packages/@react-aria/menu/src/useSafelyMouseToSubmenu.ts

```diff
@@ -0,0 +1,94 @@
+import {CSSProperties, RefObject, useEffect, useRef, useState} from 'react';
+import {useResizeObserver} from '@react-aria/utils';
+
+interface SafelyMouseToSubmenuOptions {
+  /** Ref for the parent menu. */
+  menuRef: RefObject<Element>,
+  /** Ref for the submenu. */
+  submenuRef: RefObject<Element>,
+  /** Whether the submenu is open. */
+  isOpen: boolean
```

Comment:
Do we need isOpen if we can check if submenuRef.current is defined?

Reply:
Also do we actually need menuRef?

Reply:
> Do we need isOpen if we can check if submenuRef.current is defined?

Yes, exactly. If there is a way to do it without, I'll remove it.

> Also do we actually need menuRef?

Now that the listener is on the window we don't, good catch.